src/nvim/option.c

```diff
@@ -1442,7 +1442,7 @@ do_set (
              */
             ++arg;
             if (nextchar == '&')
-              value = (intptr_t)options[opt_idx].def_val[
+              value = (long)options[opt_idx].def_val[
```

Comment:
For this reason here you probably want `(long)(intptr_t)` and above `long` should remain unchanged.